src/visual.ts

```diff
@@ -310,15 +305,19 @@ module powerbi.extensibility.visual {
                     if (y > value) {
                         value = y;
                     }
-
-                    series[valueIndex].dataPoints.push({
+                    let streamDataPoint: StreamDataPoint = {
                         x: dataPointValueIndex,
                         y: isNaN(y)
                             ? StreamGraph.DefaultValue
                             : y,
                         text: label,
                         labelFontSize: fontSizeInPx
-                    });
+                    };
+                    series[valueIndex].dataPoints.push(streamDataPoint);
+                    if (streamDataPoint.x > xMaxValue) xMaxValue = streamDataPoint.x;
```

Comment:
I would be better to move `xMaxValue = streamDataPoint.x` to the new line and use curly brackets around the statement.
Please consider it for the following statements.